### PR TITLE
Use the new parser by default

### PR DESCRIPTION
**Pull request by pablogsal**

### CODE REVIEW

**Comment by pablogsal** [comment]
> Unrelated to the PR itself, but should we send a message around on python-dev about the change to the PEP?

Ok, I will send an email to the thread on python-dev

**Comment by pablogsal** [comment]
I pushed a bunch of fixes to some tests that are failing now that pegen is the default parser (all of them are errors we discussed previously if I am not mistaken).

**Comment by lysnikolaou** [comment]
What I don't understand is why the `test_cmd_line_script` test failures weren't caught in the Ubuntu Pegen tests.

**Comment by pablogsal** [comment]
> What I don't understand is why the `test_cmd_line_script` test failures weren't caught in the Ubuntu Pegen tests.

Because they run with isolated mode (`-I`) and they ignore environment variables and also don't pass down `-p new`.

**Comment by pablogsal** [comment]
Are you ok with the latest changes? Should we merge this?

File: Python/initconfig.c [hunk 1]
```diff
@@ -646,7 +648,7 @@ config_init_defaults(PyConfig *config)
 
     config->isolated = 0;
     config->use_environment = 1;
-    config->use_peg = 0;
+    config->use_peg = 1;
```
Comment (lysnikolaou) [inline Python/initconfig.c:651]: We can probably delete this entirely now, right?

Comment (pablogsal) [reply Python/initconfig.c]: Yup! Thanks for the catch, is getting very late and I am starting to miss more things that I would like :(

Comment (lysnikolaou) [reply Python/initconfig.c]: It is indeed getting late. Already after 4 over here.

File: Python/initconfig.c [hunk 2]
```diff
@@ -39,7 +39,8 @@ Options and arguments (and corresponding environment variables):\n\
 -B     : don't write .pyc files on import; also PYTHONDONTWRITEBYTECODE=x\n\
 -c cmd : program passed in as string (terminates option list)\n\
 -d     : debug output from parser; also PYTHONDEBUG=x\n\
--p arg : use the new PEG parser, if arg is new; if arg is old, use the old\n\
+-p arg : select what parser to use: if arg is 'new' use the new PEG parser and\n\
+          if arg is 'old' use the traditional parser. Default is 'new'.\n\
```
Comment (lysnikolaou) [inline Python/initconfig.c:43]: Nit:
```suggestion
         if arg is 'old' use the traditional parser. Default is 'new'.\n\
```

**Pull request merged**